hal/common/critical.c

```diff
@@ -304,9 +294,21 @@ uint32_t core_util_atomic_decr_u32(uint32_t *valuePtr, uint32_t delta)
     return newValue;
 }
 
-void *core_util_atomic_decr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_decr((uintptr_t)valuePtr, (uintptr_t)delta);
+#endif
+
+
+bool core_util_atomic_cas_ptr(void **ptr, void **expectedCurrentValue, void *desiredValue) {
+    return (void*)core_util_atomic_cas_u32(
+            (uint32_t *)ptr,
+            (uint32_t *)expectedCurrentValue,
+            (uint32_t)desiredValue);
 }
 
-#endif
+void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta) {
```

Comment:
I cant find stddef.h included within this header to get the ptrdiff_t type, looks like its pulled in by other headers though.